test/browser/legacy_full_page_refresh_test.rb

```diff
@@ -8,6 +8,22 @@ class LegacyPagesFullPageRefreshTest < ActionDispatch::IntegrationTest
     visit "/legacy_pages/1"
   end
 
+  def wait_until_page_has(content)
+    finished_navigation = false
+    while !finished_navigation
```

Comment:
Did you try using `Capybara:DSL`? Something like `expect(page).to have_content content` should work